geocruncher/profiler/storage.py

```diff
@@ -0,0 +1,65 @@
+from abc import ABC, abstractmethod
+from pathlib import Path
+import json
+from typing import Dict, Any
+import redis
+
+
+class ProfilerStorage(ABC):
+    """Abstract base class for profiler storage backends"""
+
+    @abstractmethod
+    def save(self, computation: str, version: int, metadata: Dict[str, Any], steps: Dict[str, Dict[str, float]]) -> None:
```

Comment:
nitpick, but "Dict" is deprecated since 3.9, you can just use "dict" directly and save yourself the import
```suggestion
    def save(self, computation: str, version: int, metadata: dict[str, Any], steps: dict[str, dict[str, float]]) -> None:
```